### PR TITLE
✨ message from replication change

### PR DESCRIPTION
**Pull request by RTLS**

### CODE REVIEW

**Comment by RTLS** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/90?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#90** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/90?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#84** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/84?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 4 other dependent PRs ([#85](https://github.com/sequinstream/sequin/pull/85) <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/85?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#86](https://github.com/sequinstream/sequin/pull/86) <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/86?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#87](https://github.com/sequinstream/sequin/pull/87) <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/87?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 1 other)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RTLS and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/lib/sequin/streams/streams.ex [hunk 1]
```diff
@@ -749,4 +753,52 @@ defmodule Sequin.Streams do
   defp env do
     Application.get_env(:sequin, :env)
   end
+
+  def message_from_replication_change(%SourceTable{} = source_table, %NewRecord{} = change) do
+    user_fields =
+      Enum.map(source_table.column_mappings, fn mapping ->
+        {String.to_atom(mapping.stream_column.name), change.record[mapping.object_field]}
```
Comment (acco) [inline server/lib/sequin/streams/streams.ex:760]: keep as string?

Comment (acco) [reply server/lib/sequin/streams/streams.ex:760]: `object_field` name?

- optional selector? selector? path? 

If we wrap into bigger thing, then you would have a `row` or a `row_field` // `metadata` / `metadata_field`

File: server/lib/sequin/streams/streams.ex [hunk 2]
```diff
@@ -749,4 +753,52 @@ defmodule Sequin.Streams do
...
+      end)
+
+    [
+      data: %{
+        record: change.record,
+        old_record: nil,
+        deleted: change.type == "delete"
+      },
+      recorded_at: change.commit_timestamp
+    ] ++ user_fields
```
Comment (acco) [inline server/lib/sequin/streams/streams.ex:770]: make a struct

Comment (acco) [reply server/lib/sequin/streams/streams.ex:770]: kick `deleted` out of data

File: server/lib/sequin/streams/streams.ex [hunk 3]
```diff
@@ -749,4 +753,52 @@ defmodule Sequin.Streams do
...
+  def message_from_replication_change(%SourceTable{} = source_table, %NewRecord{} = change) do
+    user_fields =
+      Enum.map(source_table.column_mappings, fn mapping ->
+        {String.to_atom(mapping.stream_column.name), change.record[mapping.object_field]}
+      end)
+
+    [
+      data: %{
+        record: change.record,
+        old_record: nil,
```
Comment (acco) [inline server/lib/sequin/streams/streams.ex:766]: record and changes?

File: server/test/sequin/streams_test.exs [hunk 4]
```diff
@@ -914,4 +915,170 @@ defmodule Sequin.StreamsTest do
       assert mapping.object in SourceTableStreamTableColumnMapping.objects()
     end
   end
+
+  describe "message_from_replication_change/2" do
+    setup do
+      stream_table = StreamsFactory.insert_stream_table!()
```
Comment (acco) [inline server/test/sequin/streams_test.exs:921]: pass in columns here?

   columns: [
            %{name: "value", type: :integer, is_pk: false}
          ]

File: server/test/support/factory/replication_factory.ex [hunk 5]
```diff
@@ -109,4 +112,71 @@ defmodule Sequin.Factory.ReplicationFactory do
...
+  def postgres_insert(attrs \\ []) do
+    attrs = Map.new(attrs)
+
+    merge_attributes(
+      %NewRecord{
+        commit_timestamp: Factory.timestamp(),
+        errors: nil,
+        ids: [Factory.unique_integer()],
+        schema: "__postgres_replication_test_schema__",
+        table: "__postgres_replication_test_table__",
```
Comment (acco) [inline server/test/support/factory/replication_factory.ex:125]: Factory.postgres_object()

**Pull request merged**